### PR TITLE
Fix compiler specs git integration for configurable default branch

### LINKED ISSUE #11753: Compiler specs fail if git is configured with a different defaultBranch
## Bug Report
How to reproduce:

Have a .gitconfig that includes
```
[init]
	defaultBranch = main
```

In checked out crystal repo, run the compiler specs:
```
make clean
make
make compiler_spec
```
This results in 3 faileures:
```
crystal spec spec/compiler/crystal/tools/doc/project_info_spec.cr:93 # Crystal::Doc::ProjectInfo #fill_with_defaults with shard.yml git non-tagged commit
crystal spec spec/compiler/crystal/tools/doc/project_info_spec.cr:105 # Crystal::Doc::ProjectInfo #fill_with_defaults with shard.yml git non-tagged commit dirty
crystal spec spec/compiler/crystal/tools/doc/project_info_spec.cr:142 # Crystal::Doc::ProjectInfo .find_git_version

```
All three are failing due to `master` being hardcoded.

- Include Crystal compiler version (`crystal -v`) and OS. If possible, try to see if the bug still reproduces on master.

I had 1.3.1 checked out when I ran into the issue.

### PR DESCRIPTION
**Pull request by yxhuvud**
It is fairly common to define "main" as the default branch name, and
specs should not fail if that is configured.

Fixes #11753

### CODE REVIEW

File: spec/compiler/crystal/tools/doc/project_info_spec.cr [hunk 1]
```diff
@@ -7,6 +7,11 @@ private def run_git(command)
   Process.run(%(git -c user.email="" -c user.name="spec" #{command}), shell: true)
 end
 
+private def default_git_branch
+  branch_name = `git config --get init.defaultbranch`
+  branch_name == "" ? "master" : branch_name.strip
```
Comment (Sija) [inline spec/compiler/crystal/tools/doc/project_info_spec.cr:12]:
```suggestion
  branch_name.strip.presence || "master"
```

File: spec/compiler/crystal/tools/doc/project_info_spec.cr [hunk 2]
```diff
@@ -7,6 +7,11 @@ private def run_git(command)
   Process.run(%(git -c user.email="" -c user.name="spec" #{command}), shell: true)
 end
 
+private def default_git_branch
+  branch_name = `git config --get init.defaultbranch`
```
Comment (HertzDevil) [inline spec/compiler/crystal/tools/doc/project_info_spec.cr:11]: consider `Crystal::Git.git_config`

**Comment by HertzDevil** [comment]
Ideally there should be a spec for when `init.defaultbranch` is indeed overridden, but that's probably a bit difficult since it obviously cannot be overridden locally if the Git repository doesn't exist prior to `crystal init`.

**Comment by yxhuvud** [comment]
Updated based on https://github.com/crystal-lang/crystal/issues/11753#issuecomment-1016708487 . Instead of being dependent on the default branch name, it will always create a new one. There is really nothing in the behaviors being tested that care about if the branch was created by default or not.

File: spec/compiler/crystal/tools/doc/project_info_spec.cr [hunk 3]
```diff
@@ -92,23 +92,25 @@ describe Crystal::Doc::ProjectInfo do
 
       it "git non-tagged commit" do
         run_git "init"
+        run_git "checkout -b a_branch"
         run_git "add shard.yml"
         run_git "commit -m \"Initial commit\" --no-gpg-sign"
         commit_sha = `git rev-parse HEAD`.chomp
 
-        assert_with_defaults(ProjectInfo.new(nil, nil), ProjectInfo.new("foo", "master", refname: commit_sha))
+        assert_with_defaults(ProjectInfo.new(nil, nil), ProjectInfo.new("foo", "a_branch", refname: commit_sha))
```
Comment (straight-shoota) [inline spec/compiler/crystal/tools/doc/project_info_spec.cr:100]: What's the reason for changing the tested branch name tho?

**Review by straight-shoota — APPROVED** [review]

File: spec/compiler/crystal/tools/doc/project_info_spec.cr (hunk 3 above)
Comment (yxhuvud) [reply spec/compiler/crystal/tools/doc/project_info_spec.cr:100]: `checkout -b master` will give an error if the branch already exist. `-B` may be an alternative to using a different branch name, but as I'm less certain how well any output from that is printed to terminal, this seemed easier.

Comment (straight-shoota) [reply spec/compiler/crystal/tools/doc/project_info_spec.cr:100]: Output can simply be disabled with `--quiet`.

Comment (yxhuvud) [reply spec/compiler/crystal/tools/doc/project_info_spec.cr]: Right, and it turned out it wasn't needed anyhow.

**Review by straight-shoota — APPROVED** [review]

**Review by oprypin — APPROVED** [review]

**Pull request merged**